File: java-vertx/src/main/java/io/swagger/server/api/model/V3SiriProductionTimetableSubscriptionRequest.java
package io.swagger.server.api.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import io.swagger.server.api.model.V3SiriSubscriptionTopic;
import java.util.ArrayList;
import java.util.List;
import org.threeten.bp.OffsetDateTime;
import io.vertx.core.json.JsonObject;
import io.vertx.codegen.annotations.DataObject;
/**
 * V3SiriProductionTimetableSubscriptionRequest
 */

@javax.annotation.Generated(value = "io.swagger.codegen.v3.generators.java.JavaVertXServerCodegen", date = "2024-04-14T03:35:56.953023300+10:00[Australia/Sydney]")



public class V3SiriProductionTimetableSubscriptionRequest   {
  private OffsetDateTime startTime = null;

  private OffsetDateTime endTime = null;

  private String subscriberRef = null;

  private String subscriptionRef = null;

  /**
   * Siri Message Format 'xml' or 'json'
   */
  public enum SiriFormatEnum {
    NUMBER_0(0),
    NUMBER_1(1);

    private Integer value;

    SiriFormatEnum(Integer value) {
      this.value = value;
    }

    public Integer getValue() {
      return value;
    }

    @Override
    public String toString() {
      return String.valueOf(value);
    }

    public static SiriFormatEnum fromValue(Integer value) {
      for (SiriFormatEnum b : SiriFormatEnum.values()) {
        if (b.value.equals(value)) {
          return b;
        }
      }
      return null;
    }

  }
  private SiriFormatEnum siriFormat = null;

  private String siriVersion = null;

  private String consumerAddress = null;

  private OffsetDateTime initialTerminationTime = null;

  
  private List<V3SiriSubscriptionTopic> topics = new ArrayList<V3SiriSubscriptionTopic>();

  public V3SiriProductionTimetableSubscriptionRequest startTime(OffsetDateTime startTime) {
    this.startTime = startTime;
    return this;
  }

  /**
   * Siri Start Time of the Validity Period
   * @return startTime
   **/
    public OffsetDateTime getStartTime() {
    return startTime;
  }

  public void setStartTime(OffsetDateTime startTime) {
    this.startTime = startTime;
  }

  public V3SiriProductionTimetableSubscriptionRequest endTime(OffsetDateTime endTime) {
    this.endTime = endTime;
    return this;
  }

  /**
   * Siri End Time of the Validity Period
   * @return endTime
   **/
    public OffsetDateTime getEndTime() {
    return endTime;
  }

  public void setEndTime(OffsetDateTime endTime) {
    this.endTime = endTime;
  }

  public V3SiriProductionTimetableSubscriptionRequest subscriberRef(String subscriberRef) {
    this.subscriberRef = subscriberRef;
    return this;
  }

  /**
   * Siri Subscriber Ref
   * @return subscriberRef
   **/
    public String getSubscriberRef() {
    return subscriberRef;
  }

  public void setSubscriberRef(String subscriberRef) {
    this.subscriberRef = subscriberRef;
  }

  public V3SiriProductionTimetableSubscriptionRequest subscriptionRef(String subscriptionRef) {
    this.subscriptionRef = subscriptionRef;
    return this;
  }

  /**
   * Siri Subscription Ref - Unique to a Subscriber Ref
   * @return subscriptionRef
   **/
    public String getSubscriptionRef() {
    return subscriptionRef;
  }

  public void setSubscriptionRef(String subscriptionRef) {
    this.subscriptionRef = subscriptionRef;
  }

  public V3SiriProductionTimetableSubscriptionRequest siriFormat(SiriFormatEnum siriFormat) {
    this.siriFormat = siriFormat;
    return this;
  }

  /**
   * Siri Message Format 'xml' or 'json'
   * @return siriFormat
   **/
    public SiriFormatEnum getSiriFormat() {
    return siriFormat;
  }

  public void setSiriFormat(SiriFormatEnum siriFormat) {
    this.siriFormat = siriFormat;
  }

  public V3SiriProductionTimetableSubscriptionRequest siriVersion(String siriVersion) {
    this.siriVersion = siriVersion;
    return this;
  }

  /**
   * Siri Message Version '1.3' or '2.0'
   * @return siriVersion
   **/
    public String getSiriVersion() {
    return siriVersion;
  }

  public void setSiriVersion(String siriVersion) {
    this.siriVersion = siriVersion;
  }

  public V3SiriProductionTimetableSubscriptionRequest consumerAddress(String consumerAddress) {
    this.consumerAddress = consumerAddress;
    return this;
  }

  /**
   * Siri Consumer Address - Baseline and Updates will be sent to this address
   * @return consumerAddress
   **/
    public String getConsumerAddress() {
    return consumerAddress;
  }

  public void setConsumerAddress(String consumerAddress) {
    this.consumerAddress = consumerAddress;
  }

  public V3SiriProductionTimetableSubscriptionRequest initialTerminationTime(OffsetDateTime initialTerminationTime) {
    this.initialTerminationTime = initialTerminationTime;
    return this;
  }

  /**
   * Siri Initial Termination Time - Expiry of the subscription
   * @return initialTerminationTime
   **/
    public OffsetDateTime getInitialTerminationTime() {
    return initialTerminationTime;
  }

  public void setInitialTerminationTime(OffsetDateTime initialTerminationTime) {
    this.initialTerminationTime = initialTerminationTime;
  }

  public V3SiriProductionTimetableSubscriptionRequest topics(List<V3SiriSubscriptionTopic> topics) {
    this.topics = topics;
    return this;
  }

  public V3SiriProductionTimetableSubscriptionRequest addTopicsItem(V3SiriSubscriptionTopic topicsItem) {
    this.topics.add(topicsItem);
    return this;
  }

  /**
   * Get topics
   * @return topics
   **/
    public List<V3SiriSubscriptionTopic> getTopics() {
    return topics;
  }

  public void setTopics(List<V3SiriSubscriptionTopic> topics) {
    this.topics = topics;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    V3SiriProductionTimetableSubscriptionRequest v3SiriProductionTimetableSubscriptionRequest = (V3SiriProductionTimetableSubscriptionRequest) o;
    return Objects.equals(this.startTime, v3SiriProductionTimetableSubscriptionRequest.startTime) &&
        Objects.equals(this.endTime, v3SiriProductionTimetableSubscriptionRequest.endTime) &&
        Objects.equals(this.subscriberRef, v3SiriProductionTimetableSubscriptionRequest.subscriberRef) &&
        Objects.equals(this.subscriptionRef, v3SiriProductionTimetableSubscriptionRequest.subscriptionRef) &&
        Objects.equals(this.siriFormat, v3SiriProductionTimetableSubscriptionRequest.siriFormat) &&
        Objects.equals(this.siriVersion, v3SiriProductionTimetableSubscriptionRequest.siriVersion) &&
        Objects.equals(this.consumerAddress, v3SiriProductionTimetableSubscriptionRequest.consumerAddress) &&
        Objects.equals(this.initialTerminationTime, v3SiriProductionTimetableSubscriptionRequest.initialTerminationTime) &&
        Objects.equals(this.topics, v3SiriProductionTimetableSubscriptionRequest.topics);
  }

  @Override
  public int hashCode() {
    return Objects.hash(startTime, endTime, subscriberRef, subscriptionRef, siriFormat, siriVersion, consumerAddress, initialTerminationTime, topics);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class V3SiriProductionTimetableSubscriptionRequest {\n");
    
    sb.append("    startTime: ").append(toIndentedString(startTime)).append("\n");
    sb.append("    endTime: ").append(toIndentedString(endTime)).append("\n");
    sb.append("    subscriberRef: ").append(toIndentedString(subscriberRef)).append("\n");
    sb.append("    subscriptionRef: ").append(toIndentedString(subscriptionRef)).append("\n");
    sb.append("    siriFormat: ").append(toIndentedString(siriFormat)).append("\n");
    sb.append("    siriVersion: ").append(toIndentedString(siriVersion)).append("\n");
    sb.append("    consumerAddress: ").append(toIndentedString(consumerAddress)).append("\n");
    sb.append("    initialTerminationTime: ").append(toIndentedString(initialTerminationTime)).append("\n");
    sb.append("    topics: ").append(toIndentedString(topics)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: dart/lib/model/v3_siri_production_timetable_subscription_request.dart
part of swagger.api;

class V3SiriProductionTimetableSubscriptionRequest {
  /* Siri Start Time of the Validity Period */
  DateTime startTime = null;
/* Siri End Time of the Validity Period */
  DateTime endTime = null;
/* Siri Subscriber Ref */
  String subscriberRef = null;
/* Siri Subscription Ref - Unique to a Subscriber Ref */
  String subscriptionRef = null;
/* Siri Message Format 'xml' or 'json' */
  int siriFormat = null;
  //enum siriFormatEnum {  0,  1,  };
/* Siri Message Version '1.3' or '2.0' */
  String siriVersion = null;
/* Siri Consumer Address - Baseline and Updates will be sent to this address */
  String consumerAddress = null;
/* Siri Initial Termination Time - Expiry of the subscription */
  DateTime initialTerminationTime = null;

  List<V3SiriSubscriptionTopic> topics = [];

  V3SiriProductionTimetableSubscriptionRequest();

  @override
  String toString() {
    return 'V3SiriProductionTimetableSubscriptionRequest[startTime=$startTime, endTime=$endTime, subscriberRef=$subscriberRef, subscriptionRef=$subscriptionRef, siriFormat=$siriFormat, siriVersion=$siriVersion, consumerAddress=$consumerAddress, initialTerminationTime=$initialTerminationTime, topics=$topics, ]';
  }

  V3SiriProductionTimetableSubscriptionRequest.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    startTime = json['start_time'] == null ? null : DateTime.parse(json['start_time']);
    endTime = json['end_time'] == null ? null : DateTime.parse(json['end_time']);
    subscriberRef = json['subscriber_ref'];
    subscriptionRef = json['subscription_ref'];
    siriFormat = json['siri_format'];
    siriVersion = json['siri_version'];
    consumerAddress = json['consumer_address'];
    initialTerminationTime = json['initial_termination_time'] == null ? null : DateTime.parse(json['initial_termination_time']);
    topics = V3SiriSubscriptionTopic.listFromJson(json['topics']);
  }

  Map<String, dynamic> toJson() {
    return {
      'start_time': startTime == null ? '' : startTime.toUtc().toIso8601String(),
      'end_time': endTime == null ? '' : endTime.toUtc().toIso8601String(),
      'subscriber_ref': subscriberRef,
      'subscription_ref': subscriptionRef,
      'siri_format': siriFormat,
      'siri_version': siriVersion,
      'consumer_address': consumerAddress,
      'initial_termination_time': initialTerminationTime == null ? '' : initialTerminationTime.toUtc().toIso8601String(),
      'topics': topics
     };
  }

  static List<V3SiriProductionTimetableSubscriptionRequest> listFromJson(List<dynamic> json) {
    return json == null ? new List<V3SiriProductionTimetableSubscriptionRequest>() : json.map((value) => new V3SiriProductionTimetableSubscriptionRequest.fromJson(value)).toList();
  }

  static Map<String, V3SiriProductionTimetableSubscriptionRequest> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, V3SiriProductionTimetableSubscriptionRequest>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new V3SiriProductionTimetableSubscriptionRequest.fromJson(value));
    }
    return map;
  }
}
